### PR TITLE
feat: Add option to link together tokens with different decimals

### PR DESCRIPTION
**Pull request by AntonAndell**

### CODE REVIEW

**Review by mouhand999 — APPROVED** [review]

File: core-contracts/AssetManager/src/main/java/network/balanced/score/core/asset/manager/AssetManagerImpl.java [hunk 1]
```diff
@@ -45,6 +46,7 @@ public class AssetManagerImpl implements AssetManager {
     public static final String DATA_MIGRATED = "data_migrated";
```
Comment (TRSSMG) [inline core-contracts/AssetManager/src/main/java/network/balanced/score/core/asset/manager/AssetManagerImpl.java:46]: DATA_MIGRATED can be removed now

File: core-contracts/AssetManager/src/main/java/network/balanced/score/core/asset/manager/AssetManagerImpl.java [hunk 2]
```diff
@@ -316,6 +325,27 @@ private void _withdrawTo(Address asset, String from, String to, BigInteger amoun
...
+        return translateDecimals(token, amount, 1);
+    }
+
+    private BigInteger translateIncomingDecimals(String token, BigInteger amount) {
+        return translateDecimals(token, amount, -1);
+    }
+
+    private BigInteger translateDecimals(String token, BigInteger amount, int sign) {
+        BigInteger translation = decimalTransformations.get(token);
+        if (translation == null || translation.equals(BigInteger.ZERO)) {
```
Comment (TRSSMG) [inline core-contracts/AssetManager/src/main/java/network/balanced/score/core/asset/manager/AssetManagerImpl.java:338]: translation will never be equal to zero, we not not set the translation for ZERO case

**Comment by TRSSMG** [comment]
two more things: 
- When removeToken we can remove the decimalTransformations too, I think
- Don’t we have to implement translateIncomingDecimals for method withdrawRollback?

**Comment by AntonAndell** [comment]
> two more things:
> 
> * Don’t we have to implement translateIncomingDecimals for method withdrawRollback?

For the rollback i with intention did put the actual token decimals on ICON as the amount so we wouldn't need to. Please doublecheck but i think that should be covered

**Comment by TRSSMG** [comment]
all good

**Review by TRSSMG — APPROVED** [review]

**Pull request merged**